scipy/stats/_qmc.py

```diff
@@ -242,80 +253,42 @@ def discrepancy(sample, iterative=False, method='CD'):
     0.008142039609053513
 
     """
-    sample = np.asarray(sample)
+    sample = np.asarray(sample, dtype=np.float64, order="C")
```

Comment:
Is this needed? Because it's not done for `update_discrepancy`.

Reply:
Yes, it is safer to do this. This has something to do with [Row- and column-major order](https://en.wikipedia.org/wiki/Row-_and_column-major_order). It allows me to tell cython that my array is C ordered contigent, and to pass `sample` like this : 
```Python
double[:, ::1] sample
```
Which is more performant 
I forgot it in update_discrepancy, so I will add also add it, thank you @tupui 🙂